File: 250501/mypi.f90
program monte_carlo_pi
  implicit none
  integer :: i, pt_count, total_count
  real :: px, py, r_val, pi_estimate

  call random_seed()  ! Random Number Seed
  pt_count = 0
  total_count = 10000

  do i = 1, total_count
     call random_number(px)
     call random_number(py)
     r_val = sqrt(px**2 + py**2)
     if (r_val < 1.0) then
        pt_count = pt_count + 1
     end if
  end do

  pi_estimate = real(pt_count) / real(total_count) * 4.0
  print *, 'Estimated Pi = ', pi_estimate

end program monte_carlo_pi

File: 250501/mypi.c
#include <stdio.h>
#include <stdlib.h>
#include <math.h>
#include <time.h>

int main() {
    int pt_count = 0;
    int total_count = 10000;
    double px, py, r_val;

    // Random seed
    srand((unsigned int)time(NULL));

    for (int i = 0; i < total_count; i++) {
        px = (double)rand() / RAND_MAX; // 0.0 ~ 1.0 Random Number
        py = (double)rand() / RAND_MAX;
        r_val = sqrt(px * px + py * py);

        if (r_val < 1.0) {
            pt_count++;
        }
    }

    double pi_estimate = (double)pt_count / total_count * 4.0;
    printf("Estimated Pi = %f\n", pi_estimate);

    return 0;
}

